File: zemiao/104.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def maxDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if not root:
            return 0
        
        stack = [(root,1)]
        ans = 1
        while stack:
            node, depth = stack.pop()
            ans = max(ans, depth)
            if node.left:
                stack += [(node.left, depth+1)]
            if node.right:
                stack += [(node.right, depth+1)]
        
        return ans

File: jenfengchang/104.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
    int res = 0;
public:
    int maxDepth(TreeNode* root) {
        InOrderTraversal(root, 1);
        return res;
    }
    void InOrderTraversal(TreeNode* root, int lvl) {
        if(root == NULL) {
            return;
        }
        if(lvl > res) {
            res = lvl;
        }
        InOrderTraversal(root->left, lvl + 1);
        InOrderTraversal(root->right, lvl + 1);
    }
};
